lib/schema.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+module.exports = {
+  projects: {
+    route: '/accounts/{accountId}/projects',
+    requiredFields: ['accountId'],
+    id: 'projectId',
+    list: 'GET',
+    create: {
+      type: 'POST',
+      requiredFields: ['name', 'sitePrimary']
+    },
+    read: 'GET',
+    update: 'PATCH',
+    delete: 'DELETE',
+  },
+  crawls: {
+    route: '/accounts/{accountId}/projects/{projectId}/crawls',
+    requiredFields: ['accountId', 'projectId'],
+    id: 'crawlId',
+    list: 'GET',
+    create: 'POST',
+    start: 'POST',
+    read: 'GET',
+    update: 'PATCH',
+    delete: 'DELETE'
+  },
+  userAgents: {
+    route: '/user_agents',
+    id: 'userAgentCode',
+    list: 'GET',
+    read: 'GET'
+  }
```

Comment:
This is looking pretty good. A few suggestions:

- Needs a more descriptive filename. Maybe `api_schema_v2.0.0.js`?
- I would add a top-level property on the schema itself like `version: '2.0.0'` so it can be queried programmatically
- I would move projects, etc. under a top-level property called "resources" (REST terminology for the main domain objects in the API)
- Under each resource I would move the actions (list, create, etc.) into a containing property called "actions"

The containing properties add a tiny amount more nesting, but make it much clearer what is what when scanning this file, plus make it easier to iterate over the specific things you intend to without having to delete things like you're doing currently (a little weird)